src/Providers/ShortURLProvider.php

```diff
@@ -43,10 +44,17 @@ public function boot(): void
         ], 'short-url-migrations');
 
         // Routes
-        $this->loadRoutesFrom(__DIR__.'/../../routes/web.php');
+        $this->registerRoutes();
```

Comment:
Hi @ash-jc-allen 

It is necessary because of the way that the `ShortURLController` controller works.  
If the `disable_default_route` is set to true, the check [here](https://github.com/ash-jc-allen/short-url/blob/master/src/Controllers/ShortURLController.php#L30) returns `404`, because the route gets registered anyway.  
The change would allow the app that's using your package to completely override the route without setting the `disable_default_route` to true.

Here's the solution that's currently in production for my application:

```php
// On the app\Providers\AppServiceProvider
/**
 * Register any application services.
 */
public function register(): void
{
    ShortUrl::ignoreRoutes();
}
```

```php
// On the routes/web.php

// URL Shortener domain interceptor
Route::domain(config('short-url.base_url'))->group(function () {
    Route::get('{shortURLKey}', ShortURLController::class);
});
```

Reply:
Thanks for the explanation @maxacarvalho, I think I kind of get the idea. So, just to double-check I'm getting this right:

- At the moment, the package registers the controller and default route. But if the default route is disabled, a 404 is thrown in the controller.

- With your new changes, if the default route is disabled, it'll never be registered at all. So, we can remove the check from the controller method altogether?

Does that sound about right, or have I got it wrong? :)

Reply:
Hi @ash-jc-allen 

Not really, since I didn't want to mess with the `config('short-url.disable_default_route')`, I added the class `src/ShortUrl.php` and in it I added the static method `ignoreRoutes()`.  
By calling that method on the `AppServiceProvider::register` method the developer guarantees that the package's default route is not registered at all, so it can be implemented freely.  

If you see that this feature can be pushed to the config instead, and by that you realize that the controller check I mentioned above is not necessary, then we can remove the newly added class and static method in favor of the config.

Reply:
I totally understand, sorry that we couldn't get this merged in :(

Reply:
Thinking about it, I'm wondering if we could move the check into the service provider when registering the routes, like so:

```php
protected function registerRoutes(): void
{
    if (!config('short-url.disable_default_route')) {
        $this->loadRoutesFrom(__DIR__.'/../../routes/web.php');
    }
}
```

I think by doing this, we'd be able to remove these lines from the controller:

```php
/** @var Route $requestRoute */
$requestRoute = $request->route();

if ($requestRoute->getName() === 'short-url.invoke'
    && config('short-url.disable_default_route')) {
    abort(404);
}
```
From what I can see, I think that this would work. Do you see any issue with this approach? 🤔